addons/web/static/src/js/libs/moment.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * This file aim to contains the update/fix we have to do on moment.js localizations.
+ * By using updateLocale and/or defineLocale here, we avoid adding change in the official
+ * moment.js files, which could lead to conflict when updating the library.
+ */
+
+odoo.define('web.moment.extensions', function () {
+'use strict';
+const locale = moment.locale();
+moment.updateLocale('ca', {
+    preparse: function (string) {
+        return string.replace(/\b(?:d’|de )(gener|febrer|març|abril|maig|juny|juliol|agost|setembre|octubre|novembre|desembre)/g, '$1');
+    }
+});
+if(locale !== 'ca'){
+    moment.locale(locale);
+}
```

Comment:
What is purpose of this code?

Reply:
First we update the catalan localization to replace "de gener", "de febrer", ... to "gener", "febrer", ...
Then, as `moment.updateLocale('ca', ...)` will set the current localization to Catalan, we reset it to the previous local (but only if the previous wasn't the catalan language)